### ISSUE TITLE
Parsing issue for NOT without parenthesis

### ISSUE THREAD

**Issue by ThierryGrb** [sqlfluff/sqlfluff#874]
## Expected Behaviour
No error raised.

## Observed Behaviour
When not in parenthesis, the `NOT` condition is not parsed correctly.

It results in parsing error and/or rule errors (due to a wrong parsing).

```
sqlfluff lint test.sql
== [test.sql] FAIL
L:   1 | P:  21 |  PRS | Found unparsable section: ' FALSE'
```

## Steps to Reproduce
In `test.sql`:
```sql
SELECT (TRUE AND NOT FALSE)
```

Run `sqlfluff lint test.sql`.

Remarks : 
- with `SELECT TRUE AND (NOT FALSE)` there is no parsing error.
- with `SELECT TRUE AND NOT FALSE` there are rule errors, due to wrong parsing.

## Version
```
sqlfluff --version
sqlfluff, version 0.4.1
python --version
Python 3.7.9
````
## Configuration
```
[sqlfluff]
dialect = snowflake
```
(same problem with default config)

**Comment by barrywhart** [comment]
I think this is a duplicate of #827

**Comment by alanmcruickshank** [comment]
Yes. I can confirm that on current master (ed4e0561feb754ed984b3ca2ca6cf77331b0edaf) this is resolved. 👍
